diff --git a/arch/riscv/dts/fu740-c000.dtsi b/arch/riscv/dts/fu740-c000.dtsi
index 7b77c13496..9e6a7b0202 100644
--- a/arch/riscv/dts/fu740-c000.dtsi
+++ b/arch/riscv/dts/fu740-c000.dtsi
@@ -140,6 +140,39 @@
 		#size-cells = <2>;
 		compatible = "simple-bus";
 		ranges;
+		pmu {
+			compatible 			= "riscv,pmu";
+			riscv,event-to-mhpmevent =
+		/* SBI_PMU_HW_CACHE_REFERENCES -> Instruction cache/ITIM busy | Data cache/DTIM busy */
+						   <0x00003 0x00000000 0x1801>,
+		/* SBI_PMU_HW_CACHE_MISSES -> Instruction cache miss | Data cache miss or memory-mapped I/O access */
+						   <0x00004 0x00000000 0x0302>,
+		/* SBI_PMU_HW_BRANCH_INSTRUCTIONS -> Conditional branch retired */
+						   <0x00005 0x00000000 0x4000>,
+		/* SBI_PMU_HW_BRANCH_MISSES -> Branch direction misprediction | Branch/jump target misprediction */
+						   <0x00006 0x00000000 0x6001>,
+		/* L1D_READ_MISS -> Data cache miss or memory-mapped I/O access */
+						   <0x10001 0x00000000 0x0202>,
+		/* L1D_WRITE_ACCESS -> Data cache write-back */
+						   <0x10002 0x00000000 0x0402>,
+		/* L1I_READ_ACCESS -> Instruction cache miss */
+						   <0x10009 0x00000000 0x0102>,
+		/* LL_READ_MISS -> UTLB miss */
+						   <0x10011 0x00000000 0x2002>,
+		/* DTLB_READ_MISS -> Data TLB miss */
+						   <0x10019 0x00000000 0x1002>,
+		/* ITLB_READ_MISS-> Instruction TLB miss */
+						   <0x10021 0x00000000 0x0802>;
+			riscv,event-to-mhpmcounters = <0x00003 0x00006 0x18>,
+						      <0x10001 0x10002 0x18>,
+						      <0x10009 0x10009 0x18>,
+						      <0x10011 0x10011 0x18>,
+						      <0x10019 0x10019 0x18>,
+						      <0x10021 0x10021 0x18>;
+			riscv,raw-event-to-mhpmcounters = <0x0 0x0 0xffffffff 0xfc0000ff 0x18>,
+							  <0x0 0x1 0xffffffff 0xfff800ff 0x18>,
+							  <0x0 0x2 0xffffffff 0xffffe0ff 0x18>;
+		};
 		plic0: interrupt-controller@c000000 {
 			#interrupt-cells = <1>;
 			#address-cells = <0>;
